### PR TITLE
PAYARA-3422 Clean up payara pom (Payara 4.x)

### PR DESCRIPTION
**Pull request by jGauravGupta**

### CODE REVIEW

**Comment by jGauravGupta** [comment]
jenkins test please

File: appserver/batch/hazelcast-jbatch-store/pom.xml [hunk 1]
```diff
@@ -53,19 +52,17 @@
...
             <groupId>com.ibm.jbatch</groupId>
             <artifactId>com.ibm.jbatch.container</artifactId>
-            <version>${com.ibm.jbatch.container.version}</version>
             <type>jar</type>
         </dependency>
         <dependency>
             <groupId>com.hazelcast</groupId>
             <artifactId>hazelcast</artifactId>
-            <version>3.3.3</version>
+            <version>${hazelcast.version}</version>
```
Comment (smillidge) [inline appserver/batch/hazelcast-jbatch-store/pom.xml]: Does this actually need the version as it should come form the parent pom?

File: nucleus/payara-modules/notification-core/pom.xml [hunk 2]
```diff
@@ -122,7 +122,7 @@
         <dependency>
             <groupId>javax.ejb</groupId>
             <artifactId>javax.ejb-api</artifactId>
-            <version>3.2</version>
+            <version>${javax.ejb-api.version}</version>
```
Comment (smillidge) [inline nucleus/payara-modules/notification-core/pom.xml]: Does notification core actually need this dependency as it seems a bit weird.

File: nucleus/pom.xml [hunk 3]
```diff
@@ -185,6 +185,17 @@
         <javassist.version>3.24.1-GA</javassist.version>
         <javax.el.version>3.0.1-b11</javax.el.version>
         <javax.el-api.version>3.0.1-b03</javax.el-api.version>
+
+        <!-- JAX-B -->
+        <jaxb-api.version>2.2.13-b141020.1521</jaxb-api.version>
+        <jaxb.version>2.2.12-b141219.1637</jaxb.version>
+
+        <!-- EJB -->
```
Comment (smillidge) [inline nucleus/pom.xml]: I don't think EJB version should be in nucleus.

**Comment by jGauravGupta** [comment]
jenkins test please

File: nucleus/payara-modules/notification-core/pom.xml (hunk 2 above)
Comment (jGauravGupta) [reply nucleus/payara-modules/notification-core/pom.xml]: Yes, ejb-api not required in notification-core module.

File: appserver/batch/hazelcast-jbatch-store/pom.xml (hunk 1 above)
Comment (jGauravGupta) [reply appserver/batch/hazelcast-jbatch-store/pom.xml]: Updated the dependency-management of parent module.

**Comment by jGauravGupta** [comment]
jenkins test please

**Comment by jGauravGupta** [comment]
jenkins test please

**Comment by jGauravGupta** [comment]
jenkins test please

**Pull request merged**